Assets/Scripts/Core.cs

```diff
@@ -64,31 +85,42 @@ public class Core : MonoBehaviour {
             }
         }
 
-        foreach(string s in cfgTxt.Split('\n')) {
+        foreach (string s in cfgTxt.Split('\n')) {
             string[] properties = s.Split('=');
-            if(properties.Length == 2) {
+            if (properties.Length == 2) {
                 configs.Add(properties[0], properties[1]);
             }
         }
     }
 
     void FixedUpdate() {
-        if(mapRenderer.Ready) {
+        if (mapRenderer.Ready) {
             mapRenderer.FixedUpdate();
         }
     }
 
     void Update() {
-        mapDropdown.gameObject.SetActive(Cursor.lockState != CursorLockMode.Locked);
-        if(mapRenderer.Ready) {
+        //mapDropdown.gameObject.SetActive(Cursor.lockState != CursorLockMode.Locked);
+        if (mapRenderer.Ready) {
             mapRenderer.Render();
         }
+
+        RaycastHit hit;
+        var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+        if (Physics.Raycast(ray, out hit)) {
+            var target = new Vector3(Mathf.Floor(hit.point.x), hit.point.y, Mathf.Floor(hit.point.z));
+            OnRayCastHit?.Invoke(target);
+
+            if (Input.GetMouseButtonDown(0)) {
```

Comment:
This should be the first thing we check. If the mouse is not being pressed, there is no need to raycast.

Reply:
The point of raycasting first and then checking for the click is because of the `GridSelector`

Reply:
True. Forgot about that feature.